File: python_services/nlp_service.py
from flask import Flask, request, jsonify
import spacy
 
app = Flask(__name__)
 
# Cargar modelo de spaCy
nlp = spacy.load("es_core_news_sm")
 
@app.route("/analyze", methods=["POST"])
def analyze():
    #Obtener el JSON enviado desde Node.js
    data = request.get_json()
    text = data.get("text", "")
 
    if not text:
        return jsonify({"error": "Texto no proporcionado"}), 400
 
    # Analizar texto con spaCy
    doc = nlp(text)
    entities = [{"text": ent.text, "label": ent.label_} for ent in doc.ents]

    # Retornar las entidades encontradas
    return jsonify({"entities": entities})
 
if __name__ == "__main__":
    # Ejecutar el servicio en el puerto 5000
    app.run(port=5000)

File: src/services/nlpService.js
const axios = require('axios');
 
const analyzeText = async (text) => {
    try {
        // Enviar texto al servicio Flask
        const response = await axios.post('http://localhost:5000/analyze', { text });
        return response.data.entities; // Retornar las entidades encontradas por spaCy
    } catch (error) {
        console.error('Error al analizar texto con Flask:', error.message);
        throw new Error('No se pudo analizar el texto con NLP.');
    }
};
 
module.exports = { analyzeText };